heareval/predictions/task_predictions.py

```diff
@@ -781,13 +808,26 @@ def task_predictions_train(
     in_memory: bool,
     deterministic: bool,
 ) -> GridPointResult:
+    """
+    Train a predictor for a specific task using pre-computed embeddings.
+    """
+
     start = time.time()
     predictor: AbstractPredictionModel
     if metadata["embedding_type"] == "event":
-        validation_target_events = json.load(
-            embedding_path.joinpath("valid.json").open()
-        )
-        test_target_events = json.load(embedding_path.joinpath("test.json").open())
+
+        # Save the target events for validation and test
+        validation_target_events = {}
+        for split_name in data_splits["valid"]:
+            validation_target_events.update(
+                json.load(embedding_path.joinpath(f"{split_name}.json").open())
+            )
+        test_target_events = {}
+        for split_name in data_splits["test"]:
```

Comment:
Oh man this seems really confusing, particularly if you understand the other pattern from hear-preprocess that splits is a list, not a dict of lists. How do we fix this and make it clearer?

Reply:
#319